src/parse.rs

```diff
@@ -0,0 +1,269 @@
+use nom::branch::alt;
+use nom::bytes::complete::tag;
+use nom::character::complete::{char, multispace1, one_of, u64, u8};
+use nom::combinator::{opt, recognize};
+use nom::multi::{count, many1};
+use nom::sequence::{pair, preceded, terminated, tuple};
+use nom::IResult;
+
+#[test]
+fn fen_starting_board() {
+    let result = fen_to_game("rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1");
+    assert_eq!(
+        result,
+        Ok(ParsedGameState {
+            piece_position: [
+                ['r', 'n', 'b', 'q', 'k', 'b', 'n', 'r'],
+                ['p'; 8],
+                ['.'; 8],
+                ['.'; 8],
+                ['.'; 8],
+                ['.'; 8],
+                ['P'; 8],
+                ['R', 'N', 'B', 'Q', 'K', 'B', 'N', 'R']
+            ],
+            active_player: ActivePlayer::White,
+            castling_rights: CastlingRights {
+                black_kingside: true,
+                black_queenside: true,
+                white_kingside: true,
+                white_queenside: true
+            },
+            en_passant_target: None,
+            half_turn_clock: 0,
+            full_turn_clock: 1
+        })
+    );
+}
+
+#[test]
+fn fen_first_move() {
+    let result = fen_to_game("rnbqkbnr/pppppppp/8/8/4P3/8/PPPP1PPP/RNBQKBNR b KQkq e3 0 1");
+    assert_eq!(
+        result,
+        Ok(ParsedGameState {
+            piece_position: [
+                ['r', 'n', 'b', 'q', 'k', 'b', 'n', 'r'],
+                ['p', 'p', 'p', 'p', 'p', 'p', 'p', 'p'],
+                ['.', '.', '.', '.', '.', '.', '.', '.'],
+                ['.', '.', '.', '.', '.', '.', '.', '.'],
+                ['.', '.', '.', '.', 'P', '.', '.', '.'],
+                ['.', '.', '.', '.', '.', '.', '.', '.'],
+                ['P', 'P', 'P', 'P', '.', 'P', 'P', 'P'],
+                ['R', 'N', 'B', 'Q', 'K', 'B', 'N', 'R']
+            ],
+            active_player: ActivePlayer::Black,
+            castling_rights: CastlingRights {
+                black_kingside: true,
+                black_queenside: true,
+                white_kingside: true,
+                white_queenside: true
+            },
+            en_passant_target: Some((Rank::E, File::Three)),
+            half_turn_clock: 0,
+            full_turn_clock: 1
+        })
+    )
+}
```

Comment:
This is good, though we should include it in a `test` mod so it doesn't get included in builds